app/client/components/payment/update/PaymentDetailUpdate.tsx

```diff
@@ -214,6 +215,14 @@ const PaymentDetailUpdate = (props: WithProductType<ProductType>) => {
 
 	const currentPaymentMethod = subscriptionToPaymentMethod(productDetail);
 
+	const directDebitIsAllowed =
+		currentPaymentMethod == PaymentMethod.dd || (
+			productDetail.subscription.plan !== undefined &&
+			productDetail.subscription.deliveryAddress !== undefined &&
+			productDetail.subscription.plan.currency === "GBP" &&
+			productDetail.subscription.deliveryAddress.country === "United Kingdom"
```

Comment:
As part of some future work we'll be looking to remove the .plan object and rely on the currentPlans and futurePlans properties. Using getMainPlan function in shared/productResponse.ts tries to return a current plan first then falls back to any future plan or finally the .plan object. This should make it easier in the future to strip out the .plan object.

Sooooo .. this could probably become:

```
const mainPlan = getMainPlan(productDetail.subscription);
const directDebitIsAllowed = currentPaymentMethod === PaymentMethod.dd || (mainPlan && productDetail.subscription.deliveryAddress && mainPlan.currency === "GBP" && productDetail.subscription.deliveryAddress.country === "United Kingdom");
```